x/liquidity/spec/04_messages.md

```diff
@@ -1,95 +1,101 @@
+- The balance of `Depositor` does not have enough coins for `PoolCoin`
 
 ## MsgSwapWithinBatch
 
-// Swap offer coin with demand coin from the specified liquidity pool with the given order price.
+Swap coins between liquidity pools in batch with the `MsgSwapWithinBatch` message.
+
+Offer coins are swapped with demand coins for the given order price.
 
 ```go
 type MsgSwapWithinBatch struct {
-	SwapRequesterAddress string     // account address of the origin of this message
-	PoolId               uint64     // id of the liquidity pool where this message is belong to
-	SwapTypeId           uint32     // swap type id of this swap message, default 1: InstantSwap, requesting instant swap
-	OfferCoin            sdk.Coin   // offer coin of this swap message
-	DemandCoinDenom      string     // denom of demand coin of this swap message
-	OfferCoinFee         sdk.Coin   // offer coin fee for pay fees in half offer coin
-	OrderPrice           sdk.Dec    // limit order price for the order, the price is the exchange ratio of X/Y where X is the amount of the first coin and Y is the amount of the second coin when their denoms are sorted alphabetically
+    SwapRequesterAddress string     // account address of the origin of this message
+    PoolId               uint64     // id of the liquidity pool
+    SwapTypeId           uint32     // swap type id of this swap message, default 1: InstantSwap, requesting instant swap
+    OfferCoin            sdk.Coin   // offer coin of this swap
+    DemandCoinDenom      string     // denom of demand coin of this swap
+    OfferCoinFee         sdk.Coin   // offer coin fee for pay fees in half offer coin
+    OrderPrice           sdk.Dec    // limit order price where the price is the exchange ratio of X/Y where X is the amount of the first coin and Y is the amount of the second coin when their denoms are sorted alphabetically
 }
 ```
 
-**Validity check**
-
-- `MsgSwapWithinBatch` failes if
-  - `SwapRequester` address does not exist
-  - `PoolId` does not exist
-  - `SwapTypeId` does not exist
-  - denoms of `OfferCoin` or `DemandCoin` do not exist in `bank` module
-  - if the balance of `SwapRequester` does not have enough amount of coins for `OfferCoin`
-  - if `OrderPrice` <= zero
-  - if `OfferCoinFee` Equal `OfferCoin` _ `params.SwapFeeRate` _ `0.5` with truncating Int
-  - if has sufficient balance `OfferCoinFee` to reserve offer coin fee.
+## Validity checks
+
+The MsgWithdrawWithinBatch message performs validity checks. The transaction that is triggered with the `MsgSwapWithinBatch` message fails if:
+
+- `SwapRequester` address does not exist
+- `PoolId` does not exist
+- `SwapTypeId` does not exist
+- Denoms of `OfferCoin` or `DemandCoin` do not exist in `bank` module
+- The balance of `SwapRequester` does not have enough coins for `OfferCoin`
+- `OrderPrice` <= zero
+- `OfferCoinFee` equals `OfferCoin` _`params.SwapFeeRate`_ `0.5` with truncating Int
```

Comment:
This validity check is not clear to me, can you explain? what are we saying here?